### DISCUSSION TITLE
form.getInputProps.value is not equal form.values

### DISCUSSION THREAD

**Discussion by vaynevayne** [mantinedev/mantine#3918] [Q&A] [answered]
this is true ?  ` form.values.a === form.getInputProps('a').value` ? 

### What package has an issue

@mantine/form

### Describe the bug

```jsx
  const form = useForm({

    initialValues: {

        g_neo_campaign__mode: 'whereIn',
  }
  })

  console.log('...', form.values.g_neo_campaign__mode, form.getInputProps(['g_neo_campaign__mode']))


return (<></>)
```

Aren't these two ways of writing it equivalent?
```jsx
<MyComponent   value={form.getInputProps('name').value}. onChange={form.getInputProps('name').onChange}  />
!== 
<MyComponent   value={form.values.name}. onChange={val => form.setFieldValue('name',val)}  />

```

<img width="728" alt="image" src="https://user-images.githubusercontent.com/124666577/227879174-78a6a654-2bdb-4ac6-9ae1-bc4b40ea12ab.png">

### What version of @mantine/hooks page do you have in package.json?

6.0.2

### Do you know how to fix the issue

None

### Are you willing to participate in fixing this issue and create a pull request with the fix

None

**Comment by rtivital** [comment, marked as answer]
You call form.getInputProps incorrectly, first argument must be a string:
```ts
// incorrect
form.getInputProps(['g_neo_campaign__mode']);

// correct
form.getInputProps('g_neo_campaign__mode')
```

**Comment by vaynevayne** [comment]
Thanks for the correction. By the way, are these two equivalent? Which one do you recommend? @rtivital 

```jsx
// 1
<MyComponent   value={form.getInputProps('name').value}. onChange={form.getInputProps('name').onChange}  />

// 2
<MyComponent   value={form.values.name}. onChange={val => form.setFieldValue('name',val)}  />
```
